src/providers/oracle/qgsoracleprovider.cpp

```diff
@@ -3248,11 +3207,92 @@ QgsVectorLayerExporter::ExportError QgsOracleProvider::createEmptyLayer(
     QgsDebugMsgLevel( QStringLiteral( "No fields created." ), 2 );
   }
 
+  if ( !fakeColumn.isEmpty()
+       && !provider->deleteAttributes( QgsAttributeIds { provider->fields().indexOf( fakeColumn ) } ) )
+  {
+    errorMessage = QObject::tr( "Remove of temporary column '%1' failed" ).arg( fakeColumn );
+    delete provider;
```

Comment:
Can you make provider a unique_ptr instead and avoid the raw pointer handling?

Reply:
Sure. It's done